The job **`arm-android`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/17544893182/job/49824024057) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/49824024057) [(plain)](https://github.com/rust-lang/rust/commit/9571a6f9fd89e66b1486a0488099c4fc7e0ec16a/checks/49824024057/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

---- [assembly] tests/assembly-llvm/c-variadic-arm.rs stdout ----
------FileCheck stdout------------------------------

------FileCheck stderr------------------------------
/checkout/tests/assembly-llvm/c-variadic-arm.rs:16:12: error: CHECK: expected string not found in input
 // CHECK: vldr
           ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/assembly-llvm/c-variadic-arm/c-variadic-arm.s:26:12: note: scanning from here
 sub sp, sp, #8
           ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/assembly-llvm/c-variadic-arm/c-variadic-arm.s:34:2: note: possible intended match here
 ldr r2, [sp, #24]
 ^

Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/assembly-llvm/c-variadic-arm/c-variadic-arm.s
Check file: /checkout/tests/assembly-llvm/c-variadic-arm.rs

-dump-input=help explains the following input dump.

Input was:
<<<<<<
            1:  .syntax unified 
            2:  .eabi_attribute 67, "2.09" 
            3:  .eabi_attribute 6, 4 
            4:  .eabi_attribute 8, 1 
            5:  .eabi_attribute 9, 1 
            6:  .eabi_attribute 34, 0 
            7:  .eabi_attribute 15, 1 
            8:  .eabi_attribute 16, 1 
            9:  .eabi_attribute 17, 2 
           10:  .eabi_attribute 20, 1 
           11:  .eabi_attribute 21, 1 
           12:  .eabi_attribute 23, 3 
           13:  .eabi_attribute 24, 1 
           14:  .eabi_attribute 25, 1 
           15:  .eabi_attribute 38, 1 
           16:  .eabi_attribute 14, 0 
           17:  .file "c_variadic_arm.ccda00c1758a421a-cgu.0" 
           18:  .section .text.variadic,"ax",%progbits 
           19:  .globl variadic 
           20:  .p2align 2 
           21:  .type variadic,%function 
           22:  .code 32 
           23: variadic: 
           24:  .fnstart 
           25:  .pad #8 
           26:  sub sp, sp, #8 
check:16'0                X~~~~ error: no match found
           27:  .save {r11, lr} 
check:16'0     ~~~~~~~~~~~~~~~~~
           28:  push {r11, lr} 
check:16'0     ~~~~~~~~~~~~~~~~
           29:  .pad #8 
check:16'0     ~~~~~~~~~
           30:  sub sp, sp, #8 
check:16'0     ~~~~~~~~~~~~~~~~
           31:  str r2, [sp, #16] 
check:16'0     ~~~~~~~~~~~~~~~~~~~
           32:  str r3, [sp, #20] 
check:16'0     ~~~~~~~~~~~~~~~~~~~
           33:  bl __aeabi_dadd 
check:16'0     ~~~~~~~~~~~~~~~~~
           34:  ldr r2, [sp, #24] 
check:16'0     ~~~~~~~~~~~~~~~~~~~
check:16'1      ?                  possible intended match
           35:  ldr r3, [sp, #28] 
check:16'0     ~~~~~~~~~~~~~~~~~~~
           36:  bl __aeabi_dadd 
check:16'0     ~~~~~~~~~~~~~~~~~
           37:  add sp, sp, #8 
check:16'0     ~~~~~~~~~~~~~~~~
           38:  pop {r11, lr} 
check:16'0     ~~~~~~~~~~~~~~~
           39:  add sp, sp, #8 
check:16'0     ~~~~~~~~~~~~~~~~
           40:  bx lr 
check:16'0     ~~~~~~~
           41: .Lfunc_end0: 
check:16'0     ~~~~~~~~~~~~~
           42:  .size variadic, .Lfunc_end0-variadic 
check:16'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           43:  .fnend 
check:16'0     ~~~~~~~~
           44:  
check:16'0     ~
           45:  .ident "rustc version 1.91.0-nightly (9571a6f9f 2025-09-08)" 
check:16'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           46:  .section ".note.GNU-stack","",%progbits 
check:16'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>>>>>>

------------------------------------------

error: verification with 'FileCheck' failed
status: exit status: 1
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/assembly-llvm/c-variadic-arm/c-variadic-arm.s" "/checkout/tests/assembly-llvm/c-variadic-arm.rs" "--check-prefix=CHECK" "--allow-unused-prefixes" "--dump-input-context" "100"
stdout: none
--- stderr -------------------------------
/checkout/tests/assembly-llvm/c-variadic-arm.rs:16:12: error: CHECK: expected string not found in input
 // CHECK: vldr
           ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/assembly-llvm/c-variadic-arm/c-variadic-arm.s:26:12: note: scanning from here
 sub sp, sp, #8
           ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/assembly-llvm/c-variadic-arm/c-variadic-arm.s:34:2: note: possible intended match here
 ldr r2, [sp, #24]
 ^

Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/assembly-llvm/c-variadic-arm/c-variadic-arm.s
Check file: /checkout/tests/assembly-llvm/c-variadic-arm.rs

-dump-input=help explains the following input dump.

Input was:
<<<<<<
            1:  .syntax unified 
            2:  .eabi_attribute 67, "2.09" 
            3:  .eabi_attribute 6, 4 
            4:  .eabi_attribute 8, 1 
            5:  .eabi_attribute 9, 1 
            6:  .eabi_attribute 34, 0 
            7:  .eabi_attribute 15, 1 
            8:  .eabi_attribute 16, 1 
            9:  .eabi_attribute 17, 2 
           10:  .eabi_attribute 20, 1 
           11:  .eabi_attribute 21, 1 
           12:  .eabi_attribute 23, 3 
           13:  .eabi_attribute 24, 1 
           14:  .eabi_attribute 25, 1 
           15:  .eabi_attribute 38, 1 
           16:  .eabi_attribute 14, 0 
           17:  .file "c_variadic_arm.ccda00c1758a421a-cgu.0" 
           18:  .section .text.variadic,"ax",%progbits 
           19:  .globl variadic 
           20:  .p2align 2 
           21:  .type variadic,%function 
           22:  .code 32 
           23: variadic: 
           24:  .fnstart 
           25:  .pad #8 
           26:  sub sp, sp, #8 
check:16'0                X~~~~ error: no match found
           27:  .save {r11, lr} 
check:16'0     ~~~~~~~~~~~~~~~~~
           28:  push {r11, lr} 
check:16'0     ~~~~~~~~~~~~~~~~
           29:  .pad #8 
check:16'0     ~~~~~~~~~
           30:  sub sp, sp, #8 
check:16'0     ~~~~~~~~~~~~~~~~
           31:  str r2, [sp, #16] 
check:16'0     ~~~~~~~~~~~~~~~~~~~
           32:  str r3, [sp, #20] 
check:16'0     ~~~~~~~~~~~~~~~~~~~
           33:  bl __aeabi_dadd 
check:16'0     ~~~~~~~~~~~~~~~~~
           34:  ldr r2, [sp, #24] 
check:16'0     ~~~~~~~~~~~~~~~~~~~
check:16'1      ?                  possible intended match
           35:  ldr r3, [sp, #28] 
check:16'0     ~~~~~~~~~~~~~~~~~~~
           36:  bl __aeabi_dadd 
check:16'0     ~~~~~~~~~~~~~~~~~
           37:  add sp, sp, #8 
check:16'0     ~~~~~~~~~~~~~~~~
           38:  pop {r11, lr} 
check:16'0     ~~~~~~~~~~~~~~~
           39:  add sp, sp, #8 
check:16'0     ~~~~~~~~~~~~~~~~
           40:  bx lr 
check:16'0     ~~~~~~~
           41: .Lfunc_end0: 
check:16'0     ~~~~~~~~~~~~~
           42:  .size variadic, .Lfunc_end0-variadic 
check:16'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           43:  .fnend 
check:16'0     ~~~~~~~~
           44:  
check:16'0     ~
           45:  .ident "rustc version 1.91.0-nightly (9571a6f9f 2025-09-08)" 
check:16'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           46:  .section ".note.GNU-stack","",%progbits 
check:16'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>>>>>>
------------------------------------------

---- [assembly] tests/assembly-llvm/c-variadic-arm.rs stdout end ----

failures:
    [assembly] tests/assembly-llvm/c-variadic-arm.rs

test result: FAILED. 492 passed; 1 failed; 79 ignored; 0 measured; 0 filtered out; finished in 17.13s
```

</details>

--- a/compiler/rustc_codegen_llvm/src/va_arg.rs
+++ b/compiler/rustc_codegen_llvm/src/va_arg.rs
@@ -908,6 +908,21 @@ pub(super) fn emit_va_arg<'ll, 'tcx>(
             )
         }
         "aarch64" => emit_aapcs_va_arg(bx, addr, target_ty),
+        "arm" => {
+            // Types wider than 16 bytes are not currently supported. Clang has special logic for
+            // such types, but `VaArgSafe` is not implemented for any type that is this large.
+            assert!(bx.cx.size_of(target_ty).bytes() <= 16);
+
+            emit_ptr_va_arg(
+                bx,
+                addr,
+                target_ty,
+                PassMode::Direct,
+                SlotSize::Bytes4,
+                AllowHigherAlign::Yes,
+                ForceRightAdjust::No,
+            )
+        }
         "s390x" => emit_s390x_va_arg(bx, addr, target_ty),
         "powerpc" => emit_powerpc_va_arg(bx, addr, target_ty),
         "powerpc64" | "powerpc64le" => emit_ptr_va_arg(

--- a/tests/assembly-llvm/c-variadic-arm.rs
+++ b/tests/assembly-llvm/c-variadic-arm.rs
@@ -1,0 +1,26 @@
+//@ assembly-output: emit-asm
+//@ compile-flags: -Copt-level=3
+//@ only-arm
+//@ ignore-thumb
+//@ ignore-android
+#![no_std]
+#![crate_type = "lib"]
+#![feature(c_variadic)]
+
+// Check that the assembly that rustc generates matches what clang emits.
+
+#[unsafe(no_mangle)]
+unsafe extern "C" fn variadic(a: f64, mut args: ...) -> f64 {
+    // CHECK-LABEL: variadic
+    // CHECK: sub sp, sp
+
+    // CHECK: vldr
+    // CHECK: vadd.f64
+    // CHECK: vldr
+    // CHECK: vadd.f64
+    let b = args.arg::<f64>();
+    let c = args.arg::<f64>();
+    a + b + c
+
+    // CHECK: add sp, sp
+}

